cmd/notary/keys.go

```diff
@@ -449,12 +430,11 @@ func (k *keyCommander) keysRotate(cmd *cobra.Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	for _, role := range rolesToRotate {
-		if err := nRepo.RotateKey(role, k.rotateKeyServerManaged); err != nil {
-			return err
-		}
+	err = nRepo.RotateKey(rotateKeyRole, k.rotateKeyServerManaged)
+	if err == nil && k.rotateKeyServerManaged {
+		err = nRepo.Publish()
```

Comment:
Unless we were thinking about breaking `Publish` up into several publicly exposed component functions, and have the CLI call that.  It might make more sense to have the CLI call `Publish` rather than `RotateKey` call `Publish`.